### PR TITLE
Continued work over Distributed Data

### PR DESCRIPTION
**Pull request by Horusiath**
This PR brings few bug fixes, more API methods for distributed data collections and optimized, Hyperion-based serializers for ddata module (original Akka uses Protobuf).

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
Does this need to be dove-tailed with https://github.com/akkadotnet/akka.net/pull/2421 at all?

**Comment by Horusiath** [comment]
No, those are custom serializers, but they use Hyperion internally. On the other side I'm thinking, if we should fall back to proto-buf here: current version assumes, that all elements for serialized collections will use Hyperion as well, which doesn't have to be true. We might potentially run into situation where we want to use custom serializers for internal elements representation.

**Comment by Horusiath** [comment]
Ultimatelly using protobuf here is not a good option - the major point is the fact, that we have heavily generic collections here, causing a lot of pain when working with protobuf. Therefore for the time being I'm proposing to use existing serializer (a slightly optimized one Hyperion-based).

This PR is ready for review and to merge.

File: src/core/Akka/Util/Internal/StringBuilderExtensions.cs [hunk 1]
```diff
@@ -63,7 +63,8 @@ public static StringBuilder AppendJoin<T>(this StringBuilder sb, string separato
                     index++;
                     sb.Append(separator);
                     // ReSharper disable CompareNonConstrainedGenericWithNull
-                    if (enumerator.Current != null)
+                    current = enumerator.Current;
+                    if (current != null)
```
Comment (Aaronontheweb) [inline src/core/Akka/Util/Internal/StringBuilderExtensions.cs]: Nice catch

**Comment by Aaronontheweb** [comment]
@Horusiath looks good to me

**Pull request merged**